plugins/integrations/prometheus/src/main/java/org/apache/cloudstack/metrics/PrometheusExporterImpl.java

```diff
@@ -156,7 +157,7 @@ private void addHostMetrics(final List<Item> metricsList, final long dcId, final
 
             final String cpuFactor = String.valueOf(CapacityManager.CpuOverprovisioningFactor.valueIn(host.getClusterId()));
             final CapacityVO cpuCapacity = capacityDao.findByHostIdType(host.getId(), Capacity.CAPACITY_TYPE_CPU);
-            if (cpuCapacity != null) {
+            if (cpuCapacity != null && cpuCapacity.getCapacityState() == CapacityState.Enabled) {
```

Comment:
@benj-n We use 0 value to check if the host is in the maintenance. If you only show enabled hosts entries, is is not possible to track hosts in maintenance. I prefer to modify these if statement in a way that it shows zero for hosts in maintenance mode and the correct value for the reset.

Or if you really want to exclude disabled hosts entries from prometheus exporter, can you please add a global setting to control that?

Reply:
@benj-n I missed an else statement. That looks good. Ignore my comment. :-)